dd-library-php-setup.php

```diff
@@ -0,0 +1,833 @@
+        $archive = $options[OPT_TRACER_FILE];
+        echo "Copying file '${archive}' to '${tmpDirTarGz}'\n";
+        execute_or_exit("Cannot copy file from '${archive}' to '${tmpDirTarGz}'", "cp -r ${archive} ${tmpDirTarGz}");
+    } else {
+        $url = isset($options[OPT_TRACER_URL])
+            ? $options[OPT_TRACER_URL]
+            : "https://github.com/DataDog/dd-trace-php/releases/download/" .
+            $options[OPT_TRACER_VERSION] . "/datadog-php-tracer-" .
+            $options[OPT_TRACER_VERSION] . ".x86_64.tar.gz";
+        download($url, $tmpDirTarGz);
+    }
+    execute_or_exit("Cannot extract the archive", "tar -xf $tmpDirTarGz -C $tmpDir");
+
+    $installDir = $options[OPT_INSTAL_DIR] . '/' . extract_version_subdir_path($options, $tmpDir, $tmpSourcesDir);
+    $installDirSourcesDir = $installDir . '/dd-trace-sources';
+    $installDirWrapperPath = $installDirSourcesDir . '/bridge/dd_wrap_autoloader.php';
+
+    // copying sources to the final destination
+    execute_or_exit("Cannot create directory '$installDirSourcesDir'", "mkdir -p $installDirSourcesDir");
+    execute_or_exit(
+        "Cannot copy files from '$tmpSourcesDir' to '$installDirSourcesDir'",
+        "cp -r $tmpSourcesDir/* $installDirSourcesDir"
+    );
+    echo "Installed required source files to '$installDir'\n";
+
+    // Actual installation
+    foreach ($selectedBinaries as $command => $fullPath) {
+        $binaryForLog = ($command === $fullPath) ? $fullPath : "$command ($fullPath)";
+        echo "Installing to binary: $binaryForLog\n";
+        $phpProperties = ini_values($fullPath);
+
+        // Copying the extension
+        $extensionVersion = $phpProperties[PHP_API];
+
+        // Suffix (zts/debug/alpine)
+        $extensionSuffix = '';
+        if (is_alpine()) {
+            $extensionSuffix = '-alpine';
+        } elseif (is_truthy($phpProperties[IS_DEBUG])) {
+            $extensionSuffix = '-debug';
+        } elseif (is_truthy(THREAD_SAFETY)) {
+            $extensionSuffix = '-zts';
+        }
+        $extensionRealPath = $tmpExtensionsDir . '/ddtrace-' . $extensionVersion . $extensionSuffix . '.so';
+        $extensionFileName = 'ddtrace.so';
+        $extensionDestination = $phpProperties[EXTENSION_DIR] . '/' . $extensionFileName;
+
+        // Move - rename() - instead of copy() since copying does a fopen() and copies to the stream itself, causing a
+        // segfault in the PHP process that is running and had loaded the old shared object file.
+        $tmpExtName = $extensionDestination . '.tmp';
+        copy($extensionRealPath, $tmpExtName);
+        rename($tmpExtName, $extensionDestination);
+        echo "Copied '$extensionRealPath' '$extensionDestination'\n";
+
+        // Writing the ini file
+        $iniFileName = '98-ddtrace.ini';
+        $iniFilePaths = [$phpProperties[INI_CONF] . '/' . $iniFileName];
+        if (\strpos('/cli/conf.d', $phpProperties[INI_CONF]) >= 0) {
```

Comment:
you really want `!== 0`: `false >= 0` is true.

Reply:
good catch